modules/selenium/src/main/java/org/testcontainers/containers/DefaultRecordingFileFactory.java

```diff
@@ -4,20 +4,29 @@
 import java.text.SimpleDateFormat;
 import java.util.Date;
 
+import static org.testcontainers.containers.VncRecordingContainer.VncRecordingFormat;
+
 public class DefaultRecordingFileFactory implements RecordingFileFactory {
 
     private static final SimpleDateFormat filenameDateFormat = new SimpleDateFormat("YYYYMMdd-HHmmss");
     private static final String PASSED = "PASSED";
     private static final String FAILED = "FAILED";
-    private static final String FILENAME_FORMAT = "%s-%s-%s.flv";
+    private static final String FILENAME_FORMAT = "%s-%s-%s.%s";
+
 
     @Override
     public File recordingFileForTest(File vncRecordingDirectory, String prefix, boolean succeeded) {
+        return recordingFileForTest(vncRecordingDirectory, prefix, succeeded, null);
+    }
+
+    @Override
+    public File recordingFileForTest(File vncRecordingDirectory, String prefix, boolean succeeded, VncRecordingFormat recordingFormat) {
         final String resultMarker = succeeded ? PASSED : FAILED;
         final String fileName = String.format(FILENAME_FORMAT,
             resultMarker,
             prefix,
-            filenameDateFormat.format(new Date())
+            filenameDateFormat.format(new Date()),
+            VncRecordingFormat.of(recordingFormat).getExtension()
```

Comment:
Should not be `null`:
```suggestion
            recordingFormat.getExtension()
```

Reply:
Done